package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -627,33 +640,76 @@ def parse(self, **kwargs):
             fin.readline()  # y
             fin.readline()  # z
 
-            indices = np.zeros(natoms, dtype=int)
-            types = np.zeros(natoms, dtype=object)
 
+     
+            # Next line contains the headers for the atom data
             atomline = fin.readline()  # ITEM ATOMS
             attrs = atomline.split()[2:]  # attributes on coordinate line
             col_ids = {attr: i for i, attr in enumerate(attrs)}  # column ids
+            headers = {header: header in col_ids for header in DUMP_HEADERS} # Find the headers present in the dump file
```

Comment:
With this, one can loop over line `col_ids` instead of `DUMP_HEADERS` in line 675 since not all column headers in `DUMP_HEADERS` may be used. This eliminates unneeded iteration for every atom.
```suggestion
            col_ids = {attr: i for i, attr in enumerate(attrs) if attr in DUMP_HEADERS}  # column ids
            if "id" not in col_ids:
                raise ValueError("No id column found in dump file")
```